src/main/java/com/dataiku/dip/plugins/tradearea/AreaGenerator.java

```diff
@@ -0,0 +1,75 @@
+package com.dataiku.dip.plugins.tradearea;
+
+import com.dataiku.dip.pivot.backend.common.datebinner.MonthOfYearGroupedBinner;
+import com.vividsolutions.jts.geom.util.GeometryTransformer;
+
+abstract class AreaGenerator {
+
+    public abstract String generateArea(MyGeoPoint geopoint);
+
+}
+
+class RectangleAreaGenerator extends AreaGenerator {
+    double width;
+    double height;
+    double radius;
+
+    public RectangleAreaGenerator(double width, double height){
+        this.width = width;
+        this.height = height;
+        this.radius = Math.sqrt(Math.pow(width, 2)+Math.pow(height, 2));
+    }
+
+    public String generateArea(MyGeoPoint center) {
+        System.out.println("Generating an Rectangular area");
+        double startLat = center.latitude;
+        double startLong = center.longitude;
+        MyGeoPoint initGeoPoint = null;
+        MyGeoPoint tmpGeoPoint;
```

Comment:
Don't need to be declared out of the loop

Reply:
Put it inside